addon/addon.go

```diff
@@ -149,20 +151,35 @@ func NewAddonManager(kubeConfig *rest.Config, agentImage string, agentInstallAll
 		return nil, err
 	}
 
+	addonClient, err := addonv1alpha1client.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	if err != nil {
 		klog.Errorf("unable to create kube client: %v", err)
 		return addonMgr, err
 	}
 
 	agentFactory := addonfactory.NewAgentAddonFactory(AppMgrAddonName, ChartFS, ChartDir).
-		WithGetValuesFuncs(getValue, addonfactory.GetValuesFromAddonAnnotation).
+		// register the supported configuration types
+		WithConfigGVRs(
+			schema.GroupVersionResource{Group: "addon.open-cluster-management.io", Version: "v1alpha1", Resource: "addondeploymentconfigs"},
+		).
+		WithGetValuesFuncs(
+			// get the AddOnDeloymentConfig object and transform it to Values object
+			addonfactory.GetAddOnDeloymentConfigValues(
```

Comment:
I think we need put the `GetAddOnDeloymentConfigValues` at the end, like
```
WithGetValuesFuncs(
   getValue,
   addonfactory.GetValuesFromAddonAnnotation,
   addonfactory.GetAddOnDeloymentConfigValues(...)
)
```

thus we can use addondeloymentconfig to override other configs, in future, we may deprecate the addon value annotation, in that time, we need a way to replace it

thanks